From: "Barak A. Pearlmutter" <barak+git@pearlmutter.net>
Date: Thu, 6 Oct 2016 09:41:10 +0100
Subject: autoupdate

Run autoupdate, and follow its instructions.
Refresh autotools with warnings on, and heed them.
---
 Makefile.am     |  1 -
 configure.ac    | 49 ++++++++++++++++---------------------------------
 doc/Makefile.am |  2 +-
 3 files changed, 17 insertions(+), 35 deletions(-)
 mode change 100755 => 100644 configure.ac

diff --git a/Makefile.am b/Makefile.am
index 7d26cde..cd3692b 100755
--- a/Makefile.am
+++ b/Makefile.am
@@ -1,4 +1,3 @@
 ACLOCAL_AMFLAGS = -I m4
 EXTRA_DIST=copying autogen.sh readme.install
-AUTOMAKE_OPTIONS = foreign
 SUBDIRS = config src doc examples contrib misc os2 othersrc msdev m4
diff --git a/configure.ac b/configure.ac
old mode 100755
new mode 100644
index b959266..068e87d
--- a/configure.ac
+++ b/configure.ac
@@ -10,17 +10,13 @@ dnl       as those are all the systems I have access to.
 dnl ------------------------------------------------------------------------
 
 dnl Initialize Autoconf. 
-AC_INIT(pstoedit,3.70,wglunz35_AT_pstoedit.net)
+AC_INIT([pstoedit],[3.70],[wglunz35_AT_pstoedit.net])
 AC_LANG(C++)
 dnl AC_CONFIG_AUX_DIR(config)
-AM_INIT_AUTOMAKE(pstoedit,3.70)
-AM_MAINTAINER_MODE
+AM_INIT_AUTOMAKE([foreign subdir-objects -Wall])
 AC_CONFIG_SRCDIR([src/pstoedit.h])
 AC_CONFIG_SUBDIRS(libEMF-1.0)
 AC_CONFIG_MACRO_DIR([m4])
-dnl libtoolize: Consider adding `AC_CONFIG_MACRO_DIR([m4])' to configure.ac and
-dnl libtoolize: rerunning libtoolize, to keep the correct libtool macros in-tree.
-dnl libtoolize: Consider adding `-I m4' to ACLOCAL_AMFLAGS in Makefile.am.
 
 dnl We need only C++ and no other language
 dnl seems to be broken : AC_LIBTOOL_TAGS([CXX])
@@ -30,14 +26,11 @@ AC_PROG_CC
 AC_PROG_CXX
 AC_PROG_AWK
 AC_PROG_YACC
+AM_PROG_AR
 AC_PROG_INSTALL
-dnl there is not need to build the static libs if shared libs are available
-AC_DISABLE_STATIC
+dnl there is no need to build the static libs if shared libs are available
+LT_INIT([disable-static win32-dll dlopen])
 
-dnl yes, we are sure to have clean DLLs under cygwin
-AC_LIBTOOL_WIN32_DLL 
-
-AC_PROG_LIBTOOL
 AC_PROG_MAKE_SET
 
 AC_CONFIG_HEADER(src/pstoedit_config.h)
@@ -139,12 +132,10 @@ if test -n "${LIBPLOTTER_LDFLAGS}" && test -n "${LIBPLOTTER_CFLAGS}"; then
    HAVE_LIBPLOTTER=yes
    AC_MSG_CHECKING(for version of "lib plot" )   
    dnl See if we have a 2.3 or newer lib Plot version
-   AC_TRY_COMPILE( [
+   AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
 		#include <plotter.h>
 	     	typedef int Point; 
-		],[ ],
-		    AC_MSG_RESULT(you seem to have a compatible version ),
-		    [LIBPLOTTER_CFLAGS="$LIBPLOTTER_CFLAGS -DOLD_LIBPLOTTER";   
+		]], [[ ]])],[AC_MSG_RESULT(you seem to have a compatible version )],[LIBPLOTTER_CFLAGS="$LIBPLOTTER_CFLAGS -DOLD_LIBPLOTTER";   
 		    AC_MSG_RESULT(you seem to have an older version - need at least version 2.3)])
 else
    LIBPLOTTER_LDFLAGS=
@@ -166,13 +157,11 @@ CPPFLAGS=${OLD_CPPFLAGS}
 
 
 AC_MSG_CHECKING(for mkstemp)
-AC_TRY_COMPILE( [#include <stdlib.h>
+AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[#include <stdlib.h>
 		 char buffer[] = "aaaXXXXXX";
-	 	 int i = mkstemp(buffer); ],[ ],
-               [HAVE_MKSTEMP=yes;
+	 	 int i = mkstemp(buffer); ]], [[ ]])],[HAVE_MKSTEMP=yes;
                 MKSTEMP_CFLAGS="-DHAVE_MKSTEMP"; 
-                AC_MSG_RESULT(you seem to have mkstemp )], 
-               [AC_MSG_RESULT(you do not have mkstemp - so will try using tempnam instead)])
+                AC_MSG_RESULT(you seem to have mkstemp )],[AC_MSG_RESULT(you do not have mkstemp - so will try using tempnam instead)])
 
 AC_SUBST(MKSTEMP_CFLAGS)
 
@@ -311,14 +300,12 @@ else
 	   dnl See if we have a 1.0.6 or newer version
 	   OLD_CPPFLAGS=${CPPFLAGS}
 	   CPPFLAGS="$CPPFLAGS `Magick++-config --cppflags`"
-	   AC_TRY_COMPILE( [#include <Magick++.h>
+	   AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[#include <Magick++.h>
 	            #if MagickLibVersion >= 0x000008
 	 	    int correct;
 	  	    #else
 	  	    old version of GraphicsMagick;
-	  	    #endif ],[ ],
-	  	    MAGICKCORRECTVERSION="YES";AC_MSG_RESULT(you seem to have a compatible version ),
-	  	    [AC_MSG_RESULT(you seem to have an older version - need at least version 1.0.6)])
+	  	    #endif ]], [[ ]])],[MAGICKCORRECTVERSION="YES";AC_MSG_RESULT(you seem to have a compatible version )],[AC_MSG_RESULT(you seem to have an older version - need at least version 1.0.6)])
 	   CPPFLAGS=$OLD_CPPFLAGS
 	   if test -n "${MAGICKCORRECTVERSION}"; then
 	      HAVE_LIBMAGICK=yes
@@ -439,13 +426,11 @@ dnl OBS: HAVE_LIBPNG=no
 dnl OBS: LIBPNG_CFLAGS=
 dnl OBS: LIBPNG_LDFLAGS=
 dnl OBS: AC_MSG_CHECKING(See if we have libpng and zlib)
-dnl OBS: AC_TRY_COMPILE( [#include <png.h>
-dnl OBS: 	 	 int i = PNG_TRANSFORM_IDENTITY; ],[ ],
-dnl OBS:                [HAVE_LIBPNG=yes;
+dnl OBS: AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[#include <png.h>
+dnl OBS: 	 	 int i = PNG_TRANSFORM_IDENTITY; ]], [[ ]])],[dnl OBS:                HAVE_LIBPNG=yes;
 dnl OBS:                 LIBPNG_CFLAGS="-DHAVE_LIBPNG"; 
 dnl OBS:                 LIBPNG_LDFLAGS="-lpng -lz ";
-dnl OBS:                 AC_MSG_RESULT(you seem to have compatible version of libpng)], 
-dnl OBS:                [AC_MSG_RESULT(you either have no libpng or an old version - 1.0.8 should be ok.)])
+dnl OBS:                 AC_MSG_RESULT(you seem to have compatible version of libpng)],[dnl OBS:                AC_MSG_RESULT(you either have no libpng or an old version - 1.0.8 should be ok.)])
 dnl OBS: AC_SUBST(LIBPNG_LDFLAGS)
 dnl OBS: AC_SUBST(LIBPNG_CFLAGS)
 dnl OBS: AM_CONDITIONAL(HAVE_LIBPNG, test "$HAVE_LIBPNG" = yes)
@@ -456,9 +441,7 @@ dnl
 
 dnl See if we need to define a "bool" type.
 AC_MSG_CHECKING(for built-in "bool" datatype)
-AC_TRY_COMPILE(,[bool testing_bool; testing_bool=true;],
-               [CXXFLAGS="$CXXFLAGS -DINTERNALBOOL"; AC_MSG_RESULT(yes)],
-               [AC_MSG_RESULT(no)])
+AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[]], [[bool testing_bool; testing_bool=true;]])],[CXXFLAGS="$CXXFLAGS -DINTERNALBOOL"; AC_MSG_RESULT(yes)],[AC_MSG_RESULT(no)])
 
 dnl Add extra, platform-specific, C++ compiler options.
 AC_MSG_CHECKING(C++ compiler model)
diff --git a/doc/Makefile.am b/doc/Makefile.am
index 2fe0b46..65b749c 100644
--- a/doc/Makefile.am
+++ b/doc/Makefile.am
@@ -63,7 +63,7 @@ pstoedit.rtf : pstoedit.1
 	
 	
 
-clean :
+clean-local :
 	echo rm -f pstoedit.aux pstoedit.log pstoedit.dvi 
 
 realclean : clean
